### ISSUE TITLE
OneDNN verbose not visible in tensorflow 2.14.0 and above.

### ISSUE THREAD

**Issue by vishwascm** [oneapi-src/oneDNN#1758]
**OneDNN verbose not visible in tf2.14.0 and above:**
OS platform and distribution:  ubuntu 22.04.1 on aarch64

pip install tensorflow==2.14.0
export TF_ENABLE_ONEDNN_OPTS=1
export ONEDNN_VERBOSE=1
Run a example to use onednn will give following output:
oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable TF_ENABLE_ONEDNN_OPTS=0.

That's it no OneDNN logs visible.

But if I use v2.13.0
pip install tensorflow==2.13.0
export TF_ENABLE_ONEDNN_OPTS=1
export ONEDNN_VERBOSE=1
Run a example to use onednn will give following output:
Experimental oneDNN custom operations are on. If you experience issues, please turn them off by setting the environment variable TF_ENABLE_ONEDNN_OPTS=0.
onednn_verbose,info,oneDNN v2.7.3 (commit N/A)
onednn_verbose,info,cpu,runtime:threadpool,nthr:32
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,softmax_v2,ref:any,forward_inference,src_f32::blocked🔡f0 dst_f32::blocked🔡f0,,alg:softmax_accurate axis:3,1x12x2x2,0.141846.

OneDNN logs are clearly shown.

**Comment by vpirogov** [comment]
+@jondea, @milpuz01, @agramesh1

**Comment by milpuz01** [comment]
Hi @vishwascm, would you be able to share the model for which you are trying to get verbose output?

From the logs that you have shared looks like in 2.13 verbose output was coming for `softmax` layer, but in 2.14 we do not rewrite that layer to execute via oneDNN, but via Eigen (it is in this diff line 715 of PR: https://github.com/tensorflow/tensorflow/pull/60723/files#diff-65243c2e0530bb1a4ccbacb557a1803a2671551154b3ac4650688eb3729e851fL715). 

Also between 2.13 and 2.14 we have introduced new heuristics on AArch64 to decide whether to use oneDNN or Eigen library for executing primitives as there are cases when Eigen is faster then oneDNN so verbose log that you are getting in 2.13 will be different from 2.14. For more details you can see these PRs: https://github.com/tensorflow/tensorflow/pull/60026 and https://github.com/tensorflow/tensorflow/pull/61110.

Thanks,
Milos.

**Comment by vishwascm** [comment]
Hi @milpuz01, thanks for your input, as you pointed out I am using softmax layer  as below:

# Create a softmax layer
softmax_layer = tf.keras.layers.Softmax(axis=3)  

# Generate random input data
input_data = tf.random.normal((1,12,2,2))

output = softmax_layer(input_data)

My questions are:
1.  Won't enabling oneDNN (export TF_ENABLE_ONEDNN_OPTS=1) forcefully use oneDNN?
2. If not If I want to forcefully use oneDNN for benchmarking in tf v2.14.0 and above how to do it?


Regards,
Vishwas
